mmstudio/src/main/java/org/micromanager/data/internal/StorageRAM.java

```diff
@@ -121,15 +131,12 @@ public synchronized List<Image> getImagesMatching(Coords coords) {
       if (coordsToImage_ == null) {
          return null;
       }
-      List<Image> results = new ArrayList<>();
-      for (Image image : coordsToImage_.values()) {
-         // TODO figure out why subSpace was used and fix problems by not doing it
-         //  if (image.getCoords().isSubspaceCoordsOf(coords)) {
-         if (image.getCoords().equals(coords)) {
-            results.add(image);
-         }
+      try {
+         return getImagesIgnoringAxes(coords, coords.getAxes().toArray(new String[0]));
+      } catch (IOException ex) {
+         ReportingUtils.logError(ex, "Failed to read image at " + coords);
+         return null;
       }
-      return results;
```

Comment:
The reason 'subspace' (a term I made up, vaguely inspired by linear subspaces) was used is because the method was supposed to get all images that match `coords` on all the axes contained in `coords`, but the images may have additional coordinates. In other words, `getImagesMatching()` was supposed to return the images matching `coords`, ignoring axes that do _not_ appear in `coords`.

The previous code here (without the 'subspace') seems incorrect (fails to ignore axes not in `coords`), assuming `Coords.equals()` is defined as originally intended (exact match not considering axis order). Maybe this is why it is no longer used much?

I'm not sure I understand the new code -- wouldn't ignoring all the axes in `coords` mean always matching every image? Or am I missing something? (I would have thought that the axes to ignore should be those that appear in the storage but not in `coords`.)

Given that this method doesn't seem to have done what it was supposed to do for a long time, it may indeed make more  sense to deprecate it rather than totally change its behavior. Especially given that the original intended use cases of `getImagesMatching()` are now well covered by `getImagesIgnoringAxes()`.

(All this also applies to the other 2 storage implementations, of course.)